dokku

```diff
@@ -122,9 +122,9 @@ case "$1" in
 
   cleanup)
     # delete all non-running container
-    docker ps -a | grep 'Exit' | awk '{print $1}' | xargs docker rm &> /dev/null &
+    docker rm $(docker ps -a -f "status=exited" -q) &> /dev/null &
     # delete unused images
-    docker images | grep '<none>' | awk '{print $3}' | xargs docker rmi &> /dev/null &
+    docker rmi $(sudo docker images -f "dangling=true" -q) &> /dev/null &
```

Comment:
why the sudo here? I think you can remove it. 
Be careful, when there is no image to delete, you will have an error (not shown here, be the exit status will not be 0)
I use this in one of my script:

# remove exited containers
exited_containers=$(docker ps -a -q -f status=exited)
[[ -z "$exited_containers" ]] || docker rm $exited_containers &> /dev/null &

# remove dangling images
dangling_images=$(docker images -q -f dangling=true)
[[ -z "$dangling_images" ]] || docker rmi $dangling_images &> /dev/null &